.github/workflows/ci_windows.yml

```
@@ -0,0 +1,78 @@
+name: CI Windows
+
+on:
+  pull_request:
+    branches:
+      - '**'
+  workflow_dispatch:
+    inputs:
+      name:
+        description: "Manual trigger"
+  schedule:
+    - cron: '45 17 * * 6'
+
+permissions:
+  checks: write
+
+jobs:
+  build-and-tests:
+    name: Build and Tests
+    runs-on: windows-2022
+
+    env:
+      TOOLSET: v143
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
```

Same as the linux CI suggestion
 
```suggestion
      - name: Checkout
        uses: actions/checkout@v4
        with:
          fetch-depth: 0
        ref: ${{ github.ref }}
```

---

Addressed in 4117a4c9db7b0e210769cdf71a1d3ad7bc28e809